ebean-datasource-api/src/main/java/io/ebean/datasource/DataSourceConfig.java

```diff
@@ -798,6 +800,8 @@ private void loadSettings(ConfigPropertiesHelper properties) {
     offline = properties.getBoolean("offline", offline);
     shutdownOnJvmExit = properties.getBoolean("shutdownOnJvmExit", shutdownOnJvmExit);
     validateOnHeartbeat = properties.getBoolean("validateOnHeartbeat", validateOnHeartbeat);
+    closeWithinTxn = properties.get("closeWithinTxn", closeWithinTxn);
```

Comment:
unfortunately. properties does not support enums, yet. Maybe this should be changed?